### PR TITLE
Add 10k throttle to filtered index reindex query

### LINKED ISSUE #2963: Add a `requests_per_second` limit to the filtered index reindex step
## Problem

A recent run of the filtered index creation step during the data refresh pinned our Elasticsearch CPU to 99% and caused significant performance degradation on the API. We were able to mitigate this by [rethrottling the active reindex task](https://www.elastic.co/guide/en/elasticsearch/reference/8.8/docs-reindex.html#docs-reindex-rethrottle).

## Description

We will want to add a `requests_per_second` value to the `reindex` call in the ingestion server here:

https://github.com/WordPress/openverse/blob/6cdf20b70a22db30b293b34406111b44c7288e62/ingestion_server/ingestion_server/indexer.py#L506-L525

The actual value will require a bit of trial/error on our part, since the default is `-1`:

> (Optional, integer) The throttle for this request in sub-requests per second. Defaults to -1 (no throttle).

We have tried `500` with success to reduce the performance impact, but we'll want to balance that with the time it takes to complete the reindex.

Update: `500` resulted in ~10% CPU usage on ES, we've now bumped it to `2000`.

Update 2: Based on throughput (1GB at `-1`, 20MB at `500`), we estimated that ES was pushing through ~25k requests at `-1` requests/second, so we've increased the requests/second to 10k and the cluster/API have been looking reasonable.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #2963 by @AetherUnbound 

## Description

I haven't been able to test this locally due to local environment issues preventing upstream db from building and initialising correctly.

I figured I'd open the PR and see how CI handles it and get it ready ASAP in order to prevent any further delays before the next time the reindex comes around.

## Testing Instructions

Existing tests should pass. Audit Elasticsearch request logs if you want to confirm the query parameter is sent.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
LGTM, matches the ES API documentation.

**Review by obulat — APPROVED** [review]

**Review by krysal — APPROVED** [review]
LGTM!

File: ingestion_server/ingestion_server/indexer.py [hunk 1]
```diff
@@ -522,6 +522,9 @@ def create_and_populate_filtered_index(
             },
             slices="auto",
             wait_for_completion=True,
+            # 10k derived from in-production testing
+            # See https://github.com/WordPress/openverse/issues/2963
+            requests_per_second=10_000,
```
Comment (krysal) [inline ingestion_server/ingestion_server/indexer.py:527]: Actually, I'd say we should try 12k or 15k to speed up the process and prevent the timeout. It was under 50% of CPU utilization with 10k requests.
```suggestion
            requests_per_second=15_000,
```

**Comment by sarayourfriend** [comment]
Good call, @krysal, thanks.

**Pull request merged**